### ISSUE TITLE
Callback registered everytime login is called

### ISSUE THREAD

**Issue by RaphaelJenni** [EddyVerbruggen/nativescript-plugin-firebase#847]
Currently, each time you call login, a new listener gets created, what leads to that the sign in methods getting called multiple times.
https://github.com/EddyVerbruggen/nativescript-plugin-firebase/blob/master/src/firebase.android.ts#L1275

This is a) very bad performance wise, b) can have some bad side effects and c) throws errors, that are not correct (e.g. the account is already linked).

```
firebase.login(authOptions);
firebase.deleteUser();
firebase.login(authOptions);
firebase.deleteUser();
firebase.login(authOptions);

// Leads to >>
---

08-13 11:44:12.707 22057 22057 I JS      : ********************************* user.linkWithCredential(authCredential).addOnCompleteListener(onCompleteListener_1);
08-13 11:44:12.708 22057 22057 I JS      : ********************************* user.linkWithCredential(authCredential).addOnCompleteListener(onCompleteListener_1);
08-13 11:44:12.710 22057 22057 I JS      : ********************************* user.linkWithCredential(authCredential).addOnCompleteListener(onCompleteListener_1);

---

08-13 11:44:14.053 22057 22057 I JS      : Logging in the user failed. com.google.firebase.auth.FirebaseAuthUserCollisionException: This credential is already associated with a different user account.
08-13 11:44:14.366 22057 22057 I JS      : Logging in the user failed. com.google.firebase.auth.FirebaseAuthUserCollisionException: This credential is already associated with a different user account.

```

**Comment by RaphaelJenni** [comment]
Only have tried it on Android. Not sure about iOS, but I can try it later. I register with an anonymous account and then link it to a google account.

**Comment by EddyVerbruggen** [comment]
Thanks for the details!

I think it only occured on Android as the login sequence was registering `application.on(<eventname>, <callback>)` listeners but never `.off`'d them.

That's fixed in 6.5.0.
